apis/python/src/tiledbsc/soma.py

```diff
@@ -464,6 +471,104 @@ def write_raw_group(self, raw_group_uri: str, raw: ad.Raw):
+
+        :param uns_group_uri: URI where the group is to be written.
+        :param uns: anndata.uns.
+        """
+
+        if self.verbose:
+            s = util.get_start_stamp()
+            print(f"    START  WRITING {uns_group_uri}")
+
+        tiledb.group_create(uns_group_uri, ctx=self.ctx)
+        uns_group = tiledb.Group(uns_group_uri, mode="w", ctx=self.ctx)
+
+        for key in uns.keys():
+            component_uri = os.path.join(uns_group_uri, key)
+            value = uns[key]
+
+            if key == 'rank_genes_groups':
+                # TODO:
+                # This is of type 'structured array':
+                # https://numpy.org/doc/stable/user/basics.rec.html
+                #
+                # >>> a.uns['rank_genes_groups']['names'].dtype
+                # dtype([('0', 'O'), ('1', 'O'), ('2', 'O'), ('3', 'O'), ('4', 'O'), ('5', 'O'), ('6', 'O'), ('7', 'O')])
+                # >>> type(a.uns['rank_genes_groups']['names'])
+                # <class 'numpy.ndarray'>
+                #
+                # We don’t have a way to model this directly in TileDB schema right now. We support
+                # multiplicities of a single scalar type, e.g. a record array with cell_val_num==3
+                # and float32 slots (which would correspond to numpy record array
+                # np.dtype([("field1", "f4"), ("field2", "f4"), ("field3", "f4",)])). We don’t
+                # support nested cells, AKA "list" type.
+                print("      Skipping structured array:", component_uri)
+                continue
+
+            if isinstance(value, dict) or isinstance(value, ad.compat.OverloadedDict):
+                # Nested data, e.g. a.uns['draw-graph']['params']['layout']
+                self.write_uns_group(component_uri, value)
+
+            elif isinstance(value, pd.DataFrame):
+                self.write_uns_pandas_dataframe(component_uri, value)
+
+            elif isinstance(value, scipy.sparse.csr_matrix):
+                self.write_uns_scipy_sparse_csr_matrix(component_uri, value)
+
+            elif util.is_numpyable_object(value):
+                if self.verbose:
+                    s2 = util.get_start_stamp()
+                    print(f"      START  WRITING NUMPY {component_uri}")
+                util.numpyable_object_to_tiledb_array(value, component_uri, self.ctx)
+                if self.verbose:
+                    print(util.format_elapsed(s2, f"      FINISH WRITING NUMPY {component_uri}"))
+
+            else:
+                print("      Skipping unrecognized type:", component_uri, type(value))
+                continue
+
+            uns_group.add(uri=component_uri, relative=False, name=key)
+
+        uns_group.close()
+
+        if self.verbose:
+            print(util.format_elapsed(s, f"    FINISH WRITING {uns_group_uri}"))
+
+
+    # ----------------------------------------------------------------
+    def write_uns_pandas_dataframe(self, array_uri, array: pd.DataFrame):
```

Comment:
Any reason this arg is named `array` rather than `dataframe`?

Reply:
no _good_ reason @aaronwolen ;)